File: sharedlife-backend/src/main/java/net/tfg/sharedlife/dto/NewUserDto.java
package net.tfg.sharedlife.dto;

import java.util.HashSet;
import java.util.Set;

import lombok.Data;

// DTO que envia la informacion necesaria para la creación de una usuario
@Data
public class NewUserDto {
	private String firstName;
	private String lastName;
	private String email;
	private String password;
	private String username;
	private Set<String> roles = new HashSet<>();
}

File: sharedlife-frontend/src/app/models/user/new-user-dto/NewUserDTO.ts
export class NewUserDTO {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  username: string;
  //isOwner: boolean;
  roles: string[];

  constructor(fn?: string, ln?: string, e?: string, p?: string, un?: string){
    this.firstName = fn ?? "";
    this.lastName = ln ?? "";
    this.email = e ?? "";
    this.password = p ?? "";
    this.username = un ?? "";
    this.roles = [];
  }
}